src/dxil/wrapper.rs

```diff
@@ -0,0 +1,480 @@
+#![allow(
+    clippy::too_many_arguments,
+    clippy::new_without_default,
+    clippy::type_complexity
+)]
+
+use crate::dxil::ffi::*;
+use crate::ffi::*;
+use crate::os::HRESULT;
+use crate::utils::HassleError;
+use com_rs::ComPtr;
+use libloading::{Library, Symbol};
+use std::ffi::CStr;
+use winapi::um::{d3d12shader, d3dcommon};
+
+#[derive(Debug)]
+pub struct Dxil {
+    dxil_lib: Library,
+}
+
+impl Dxil {
+    pub fn new() -> Result<Self, HassleError> {
+        let dxil_lib = Library::new("dxil.dll").map_err(|e| HassleError::LoadLibraryError {
+            filename: "dxil".to_string(),
+            inner: e,
+        })?;
```

Comment:
Ik snap niet helemaal waarom we deze hebben:
```rust
    #[error("Error loading library {filename:?} / {inner:?}")]
    LoadLibraryError {
        filename: String,
        inner: libloading::Error,
    },
```

Terwijl er direct daaronder een mooie `#[from]` variant staat die niet gebruikt word:

```rust
    #[error("LibLoading error: {0:?}")]
    LibLoadingError(#[from] libloading::Error),
```
Is niet onderdeel van deze PR, maar misschien kun je ze samenvoegen? Volgens mij is het pad nodig omdat Windows alleen een nutteloze "library not found" error print. Maar dan is het alsnog fijn om maar één enum variant te hebben met een echte `#[from]`.